### DISCUSSION TITLE
sqlitestudio is very hard to actually build

### DISCUSSION THREAD

**Discussion by martinrotter** [pawelsalawa/sqlitestudio#4347] [General]
I mean, I am quite experienced programmer, hello my fellow code wizard, but this is just nonsense.

Your [build instructions](https://github.com/pawelsalawa/sqlitestudio/wiki/Instructions_for_compilation_under_Windows) are so ridiculously long, that I gave up very early:

1. Overally, instructions and whole process is just very cumbersome.
2. "Source code layout" idea is just plain bad, forcing user to use some "hardcoded" file/dir structure. You should use build system variables instead for this like "SQLITE_ROOT_DIR", "ZLIB_ROOT_DIR" etc. The way you have it user must tweak existing directory structures just to fit this single project.
3. Your compilation is somehow mingw-specific while qmake abstracts this away - that is its primary goal. Your application should be compilable with MSVC without any problems.
4. Your qmake/code contains instructions incompatible with MSVC like
```
windowscrashhandler.h:45: error: C2039: 'unexpected_handler': is not a member of 'std'
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\map(25): note: see declaration of 'std'
```
5. Also, invalid compiler arguments like `-pedantic` are passed when not really needed.
6. Is there a way to get "development" CI build installers/zips to have latest features/fixes? I need binaries with https://github.com/pawelsalawa/sqlitestudio/issues/4207 fixed.

Thank you, your app is good.

**Comment by pawelsalawa** [comment]
I'm aware of all issues you'e mentioned and I acknowledge them. I could explain why it is how it is, but let's not waste words on that. It's bad and I agree. It's a technical debt to make things right about the build process, for which there was never time, since I have very little time to spend on the project these days and these short pieces that I have I spend on bugfixing/coding features, as this brings the most value to end users.

Regarding MSVC - I initially wrote the app under Linux/GCC and then just compiled it with mingw. I've made an attempt to build it with MSVC some time ago and I gave up after 3 days of trying.

I think that at the moment the best way to understand how to build the project is to follow Github Action workflow file.

I have also updated workflow files to trigger all builds for all platforms during the night, so you could fetch nightly builds from the actions page, starting with upcoming night and assuming everything will build correctly, as bleeding edge is never certiain...

**Reply by martinrotter** [reply]
Btw, I have found those CI files, at least something. :)
